.github/workflows/fuzzy-ci.yml

```diff
@@ -0,0 +1,475 @@
+      uses: actions/cache@v3
+      id: irmin-cache
+      with:
+        path: irmin/
+        key: os${{ runner.os }}+arch${{ runner.arch }}+${{ hashFiles('fuzzy-ci-helpers/irmin.3.9.0.opam.locked') }}+${{ env.IRMIN_VERSION }}+${{ env.COMPILER_VERSION }}
+
+    - name: Download Irmin tarball
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: |
+        wget https://github.com/mirage/irmin/releases/download/$IRMIN_VERSION/irmin-$IRMIN_VERSION.tbz
+
+    - name: Create irmin dir
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: mkdir -p irmin
+
+    - name: Decompress Irmin tarball
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: tar xvf irmin-$IRMIN_VERSION.tbz -C irmin --strip-components=1
+
+    - name: Get Irmin's lock files
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: |
+        cp .github/fuzzy-ci-helpers/irmin.3.9.0.opam.locked irmin/irmin.opam.locked
+
+    - name: Install opam monorepo
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: opam install opam-monorepo --yes
+
+    - name: Pull in Irmin's dependencies
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: |
+        git checkout ${{ github.sha }}
+        opam monorepo pull --lockfile=irmin.opam.locked --yes
+      working-directory: irmin
+
+    - name: Prune Irmin
+      if: steps.irmin-cache.outputs.cache-hit != 'true'
+      run: |
+        rm -r examples/ bench/
+        find test/ -mindepth 1 -maxdepth 1 -type d -not -name 'irmin-pack' -exec rm -r {} \;
+        find src/ -mindepth 1 -maxdepth 1 -type d \
+        -not -name 'irmin-pack' \
+        -not -name 'irmin' \
+        -not -name 'irmin-tezos' \
+        -not -name ppx_irmin \
+        -not -name irmin_test \
+        -not -name irmin-test \
+        -exec rm -r {} \;
+      working-directory: irmin
+
+    - name: Build Irmin
+      run: |
+        opam exec -- dune build @check
+      working-directory: irmin
+
+    - name: Pull merl-an from cache if possible
+      uses: actions/cache@v3
+      id: merl-an-cache
+      with:
+        path: /usr/local/bin/merl-an
+        key: os${{ runner.os }}+arch${{ runner.arch }}+merl-an-sha$MERL_AN_SHA
+
+    - name: Install merl-an
+      if: steps.merl-an-cache.outputs.cache-hit != 'true'
+      run: opam pin -y merl-an https://github.com/pitag-ha/merl-an.git#$MERL_AN_SHA
+
+    - name: Add merl-an to /usr/local/bin/
+      if: steps.merl-an-cache.outputs.cache-hit != 'true'
+      run: opam exec -- cp $GITHUB_WORKSPACE/_opam/bin/merl-an /usr/local/bin/merl-an
+
+    - name: Create data set of Merlin responses
+      run: |
+        opam exec -- merl-an behavior \
+        --queries=type-enclosing,occurrences,locate,complete-prefix,errors \
+        --sample-size=30 \
+        --data=${{ env.data_dir }} \
+        --merlin=ocamlmerlin \
+        --project=irmin/src/irmin,irmin/src/irmin-pack,irmin/test/irmin-pack
```

Comment:
Yes, adding more diverse projects to the workflow should be relatively easy (I'd then decrease the number of samples per file, though, to avoid having the CI run for too long). Do you have projects in mind?

Reply:
I would target something that is part of Irmin's dependencies but didn't have a look yet (and probably won't before next year...).

Reply:
We'd need to have a look if anything in the Irmin dependency cone adds additional value to the test base. As mentioned in [the comment about the CI execution time](https://github.com/ocaml/merlin/pull/1716#discussion_r1435205285), it's better for the execution time to have few files with lots of samples than to have lots of files with few samples.

> (and probably won't before next year...).

(same for me :) )